jscomp/runtime/js_string.ml

```diff
@@ -106,326 +101,363 @@ external get : t -> int -> t = ""
 
-external indexOf : t -> int = "indexOf"
-[@@mel.send.pipe: t]
+external indexOf : t -> sub:t -> int = "indexOf"
+[@@mel.send]
 (**
-  [indexOf searchValue s] returns the position at which [searchValue] was first found within [s], or [-1] if [searchValue] is not in [s].
+  [indexOf s ~sub:searchValue ] returns the position at which [searchValue] was
+  first found within [s], or [-1] if [searchValue] is not in [s].
 
 {[
-  indexOf "ok" "bookseller" = 2;;
-  indexOf "sell" "bookseller" = 4;;
-  indexOf "ee" "beekeeper" = 1;;
-  indexOf "xyz" "bookseller" = -1;;
+  indexOf "bookseller" ~sub:"ok" = 2;;
+  indexOf "bookseller" ~sub:"sell" = 4;;
+  indexOf "beekeeper" ~sub:"ee" = 1;;
+  indexOf "bookseller" ~sub:"xyz" = -1;;
 ]}
 *)
 
-external indexOfFrom : t -> int -> int = "indexOf"
-[@@mel.send.pipe: t]
-(**
-  [indexOfFrom searchValue start s] returns the position at which [searchValue] was found within [s] starting at character position [start], or [-1] if [searchValue] is not found in that portion of [s]. The return value is relative to the beginning of the string, no matter where the search started from.
+external indexOfFrom : t -> sub:t -> pos:int -> int = "indexOf"
+[@@mel.send]
+(** [indexOfFrom s ~sub:searchValue ~start] returns the position at which
+    [searchValue] was found within [s] starting at character position [start],
+    or [-1] if [searchValue] is not found in that portion of [s]. The return
+    value is relative to the beginning of the string, no matter where the
+    search started from.
 
 {[
-  indexOfFrom "ok" 1 "bookseller" = 2;;
-  indexOfFrom "sell" 2 "bookseller" = 4;;
-  indexOfFrom "sell" 5 "bookseller" = -1;;
-  indexOf "xyz" "bookseller" = -1;;
+  indexOfFrom "bookseller" ~sub:"ok" ~pos:1 = 2;;
+  indexOfFrom "bookseller" ~sub:"sell" ~pos:2 = 4;;
+  indexOfFrom "bookseller" ~sub:"sell" ~pos:5 = -1;;
 ]}
 *)
 
-external lastIndexOf : t -> int = "lastIndexOf"
-[@@mel.send.pipe: t]
+external lastIndexOf : t -> sub:t -> int = "lastIndexOf"
+[@@mel.send]
 (**
-  [lastIndexOf searchValue s] returns the position of the {i last} occurrence of [searchValue] within [s], searching backwards from the end of the string. Returns [-1] if [searchValue] is not in [s]. The return value is always relative to the beginning of the string.
+  [lastIndexOf s ~sub:searchValue ] returns the position of the {i last}
+  occurrence of [searchValue] within [s], searching backwards from the end of
+  the string. Returns [-1] if [searchValue] is not in [s]. The return value is
+  always relative to the beginning of the string.
 
 {[
-  lastIndexOf "ok" "bookseller" = 2;;
-  lastIndexOf "ee" "beekeeper" = 4;;
-  lastIndexOf "xyz" "abcdefg" = -1;;
+  lastIndexOf "bookseller" ~sub:"ok" = 2;;
+  lastIndexOf "beekeeper" ~sub:"ee" = 4;;
+  lastIndexOf "abcdefg" ~sub:"xyz" = -1;;
 ]}
 *)
 
-external lastIndexOfFrom : t -> int -> int = "lastIndexOf"
-[@@mel.send.pipe: t]
+external lastIndexOfFrom : t -> sub:t -> pos:int -> int = "lastIndexOf"
```

Comment:
Same comment about simplifying.

Reply:
so turns out that `lastIndexOf` is one that we can't simplify like the others! the undefined argument influences the result.